esmvaltool/preprocessor/_multimodel.py

```diff
@@ -113,12 +116,20 @@ def _put_in_cube(template_cube, cube_data, stat_name,
     lons = template_cube.coord('longitude')
 
     # no plevs
+
     if len(template_cube.shape) == 3:
         cspec = [(times, 0), (lats, 1), (lons, 2)]
     # plevs
     elif len(template_cube.shape) == 4:
         plev = template_cube.coord('air_pressure')
         cspec = [(times, 0), (plev, 1), (lats, 2), (lons, 3)]
+    elif len(template_cube.shape) == 1:
+        cspec = [(times, 0), ]
+    elif len(template_cube.shape) == 2:
+        # If you're going to hardwire air_pressure into this,
+        # might as well have depth here too.
+        plev = template_cube.coord('depth')
+        cspec = [(times, 0), (plev, 1), ]
```

Comment:
See above comment.

Reply:
Hi, to be honest, this whole section could use some work (outside of this PR). For one, it appears that both the 'air_pressure' and 'depth' dimension names are hardwired into the preprocessor. Also, it appears that the four conditions are only linked with the number of dimensions, instead of which dimensions are present in the cube.

I'm guessing @valeriupredoi may want to address this in the future?